### PR TITLE
Iz cleanup

### PR DESCRIPTION
**Pull request by zarsky-broad**
* export pre-hyperscript-wrapped components
* use jsdoc for param documentation
* get jest to a workable state
* replace underscore with lodash
* code cleanup

### CODE REVIEW

File: config-overrides.js [hunk 1]
```diff
@@ -3,7 +3,14 @@ const rewireReactHotLoader = require('react-app-rewire-hot-loader')
...
 /* config-overrides.js */
-module.exports = function override(config, env) {
-  config = mixinDeep(config, manualOverrides)
-  return rewireReactHotLoader(config, env)
+module.exports = {
+  webpack: function override(config, env) {
+    config = mixinDeep(config, manualOverrides)
+    return rewireReactHotLoader(config, env)
+  },
+  jest: function(config) {
```
Comment (zarsky-broad) [inline config-overrides.js:11]: Jest (the test runner) has a separate config; this lets it read our non-relative module paths

File: src/components/common.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 import Interactive from 'react-interactive'
-import _ from 'underscore'
 import { icon } from 'src/components/icons'
 import * as Nav from 'src/libs/nav'
 import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
 
 
-const link = function(props, children) {
+export const link = function(props, children) {
```
Comment (zarsky-broad) [inline src/components/common.js:13]: In-place exports easily and clearly show which things are public

File: src/components/common.js [hunk 3]
```diff
@@ -65,111 +65,114 @@ const search = function({ wrapperProps = {}, inputProps = {} }) {
...
 
-const topBar = children => h(TopBarConstructor, children)
-
-class TopBarConstructor extends Component {
-  constructor(props) {
-    super(props)
-    this.state = { navShown: false }
-  }
+export function topBar(title, children) {
+  return h(class topBar extends Component {
```
Comment (zarsky-broad) [inline src/components/common.js]: Exporting a function like this means that our components are called in the same way as non-custom components, and can still be documented & show up with the same component name in code and in the react devtools.

File: src/components/common.js [hunk 4]
```diff
@@ -65,111 +65,114 @@ const search = function({ wrapperProps = {}, inputProps = {} }) {
...
-                textDecoration: 'none', display: 'block'
-              },
-              href: Nav.getLink('workspaces'),
-              onClick: hideNav
-            }, [
-              icon('grid-view', { class: 'is-solid', style: { margin: '0 1rem 0 1rem' } }),
-              'Projects'
-            ])
+              style: { flexGrow: 1, cursor: 'pointer' },
+              onClick: () => this.hideNav()
```
Comment (zarsky-broad) [inline src/components/common.js]: Within the containing function wrapper, we need to explicitly call methods that are bound in order to maintain the value of `this` as the instance.

File: src/components/common.js [hunk 5]
```diff
@@ -65,111 +65,114 @@ const search = function({ wrapperProps = {}, inputProps = {} }) {
...
+export const topBar = function(title, children) {
+  return h(class topBar extends Component {
+    constructor(props) {
+      super(props)
+      this.state = { navShown: false }
+    }
 
-  render() {
-    const showNav = () => {
+    showNav() {
```
Comment (zarsky-broad) [inline src/components/common.js]: Methods on the class don't have to be recreated every render

File: src/components/table.js [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
-* itemsPerPage:             required
-* itemsPerPageOptions:      required
-*/
-const paginator = function({
-                             filteredDataLength, pageNumber, setPageNumber, setItemsPerPage,
-                             itemsPerPage, itemsPerPageOptions
-                           }) {
+
+/**
+ * @callback handleNewVal
```
Comment (zarsky-broad) [inline src/components/table.js]: This is now a callback format that can be referenced in any other jsdoc block

File: src/components/common.js [hunk 7]
```diff
@@ -65,111 +65,114 @@ const search = function({ wrapperProps = {}, inputProps = {} }) {
...
 }
 
-const topBar = children => h(TopBarConstructor, children)
-
-class TopBarConstructor extends Component {
-  constructor(props) {
-    super(props)
-    this.state = { navShown: false }
-  }
+export const topBar = function(title, children) {
```
Comment (zarsky-broad) [inline src/components/common.js]: Exporting a function like this means that our components are called in the same way as non-custom components, and can still be documented & show up with the same component name in code and in the react devtools

File: config-overrides.js (hunk 1 above)
Comment (dmohs) [reply config-overrides.js:11]: I think this excellent explanation belongs in the code rather than this PR.

File: src/components/common.js (hunk 7 above)
Comment (dmohs) [reply src/components/common.js]: Curious: Does this cause the class to be redefined on every call?

File: src/pages/workspaces/List.test.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import ReactDOM from 'react-dom'
+import { h } from 'react-hyperscript-helpers'
+import { WorkspaceList } from './List'
+
+
+jest.mock('src/libs/ajax')
+
+test('renders the workspace list', () => {
+  const div = document.createElement('div')
+  ReactDOM.render(h(WorkspaceList), div)
```
Comment (dmohs) [inline src/pages/workspaces/List.test.js]: If the only value here is the absence of an error that should be noted.

File: src/components/table.js [hunk 9]
```diff
@@ -40,120 +46,112 @@ const paginator = function({
...
+            paginator({
+              filteredDataLength: dataSource.length,
+              setPageNumber: (n => {
+                this.setState({ pageNumber: n })
+                _.attempt(onPageChanged, n)
+              }),
+              pageNumber,
+              setItemsPerPage: allowItemsPerPage ? (n => {
+                this.setState({ itemsPerPage: n })
+                _.attempt(onItemsPerPageChanged, n)
```
Comment (dmohs) [inline src/components/table.js]: Explain offline please.

File: src/components/table.js [hunk 10]
```diff
@@ -1,22 +1,28 @@
...
-                             itemsPerPage, itemsPerPageOptions
-                           }) {
+
+/**
+ * @callback handleNewVal
+ * @param newValue
+ */
+
+/**
+ * @param {number} props.filteredDataLength
```
Comment (dmohs) [inline src/components/table.js:13]: Please demo this to me offline.

**Review by dmohs — APPROVED** [review]

File: src/components/common.js (hunk 7 above)
Comment (zarsky-broad) [reply src/components/common.js]: Yes, yes it did!

**Pull request merged**